File: 剑指offer/42.连续子数组的最大和.py
from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        pre = 0
        cur = 0

        ans = 0

        for num in nums:
            cur = num + max(pre, 0)
            ans = max(ans, cur)
            pre = cur

        return ans

File: 剑指offer/42.连续子数组的最大和.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums)
    {
        int pre = -9999;
        int ans = pre;

        for (int num : nums) {
            int cur = max(num, pre + num);
            pre = cur;
            ans = max(ans, pre);
        }

        return ans;
    }
};